src/component/container/page_list.tsx

```diff
@@ -11,14 +11,109 @@ export interface PageListProps {
 	store: Store;
 }
 
+export interface PageListItemProps {
+	id: string;
+	name: string;
+	pageRef: PageRef;
+	store: Store;
+}
+
+@observer
+export class PageListItem extends React.Component<PageListItemProps> {
+	@MobX.observable protected pageElementEditable: boolean = false;
+	@MobX.observable protected pageName: string;
+
+	public constructor(props: PageListItemProps) {
+		super(props);
+
+		this.handleInputChange = this.handleInputChange.bind(this);
+		this.handleBlur = this.handleBlur.bind(this);
+		this.handlePageKeyDown = this.handlePageKeyDown.bind(this);
+		this.handlePageClick = this.handlePageClick.bind(this);
+		this.handlePageDoubleClick = this.handlePageDoubleClick.bind(this);
+		this.renamePage = this.renamePage.bind(this);
+	}
+	public render(): JSX.Element {
+		return (
+			<DropdownItemEditableLink
+				editable={this.pageElementEditable}
+				focused={this.pageElementEditable}
+				handleChange={this.handleInputChange}
+				handleClick={(e: React.MouseEvent<HTMLElement>) => {
+					e.preventDefault();
+					this.handlePageClick(this.props.id);
+				}}
+				handleDoubleClick={this.handlePageDoubleClick}
+				handleKeyDown={this.handlePageKeyDown}
+				name={this.props.name}
+				handleBlur={this.handleBlur}
+				placeholder={this.props.name}
+				value={this.pageName ? this.pageName : this.props.name}
+			/>
+		);
+	}
+
+	protected handlePageClick(id: string): void {
+		this.props.store.openPage(id);
```

Comment:
you can just use `this.props.id` here. no need to pass it as an argument since you're already only responsible for one item at a time

Reply:
Done ✅